src/server/routes/api/document/pubmed/searchPubmed.js

```diff
@@ -38,31 +56,109 @@ const checkEsearchResult = json => {
 const eSearchPubmed = ( term, opts ) => {
   const params = _.assign( {}, DEFAULT_ESEARCH_PARAMS, { term }, opts );
   const url = EUTILS_SEARCH_URL + '?' + queryString.stringify( params );
-  const userAgent = `${process.env.npm_package_name}/${process.env.npm_package_version}`;
-  return fetch( url, {
-    method: 'GET',
-    headers: {
-      'User-Agent': userAgent
+  return safeFetch( url )
+    .then( checkEsearchResult ) // Error (programmatic)
+    .then( pubmedDataConverter );
+};
+
+/**
+ * Search for an article using Pubmed and CrossRef
+ *
+ * This approach uses a multi-stage approach:
+ *   1. Query the PubMed EUTIL service
+ *   2. If no unique hit, query Crossref API works resource component
+ *   3. Send the CrossRef-derived DOI to PMC to map to a PMID
+ *   4. If no PMID, try the Pubmed EUTILS service with DOI
+ *
+ * @param {String} term The search term to find the article (e.g. title)
+ * @param {String} IdType One of ID_TYPE
+ * @param {Object} opts Pubmed API options
+ * @returns {Object} A Pubmed search result object
+ */
+const crSearchPubmed = async ( term, IdType, opts ) => {
+
+  // try PubMed ESEARCH
+  const queryOpts = IdType === ID_TYPE.TITLE ? { field: ID_TYPE.TITLE } : {};
+  const searchResponse = await eSearchPubmed( term, queryOpts );
+  if( searchResponse.count === 1 ){
```

Comment:
Would it be worthwhile to also check if the first result has an exact match of the title?  There may be edge cases where we get more than one result, but the top result is still the right match.

Reply:
That's doable: search through top 10 hits?

Reply:
Yeah, that's more robust than just looking at the top one

Reply:
Do you have a test case for this?

Reply:
No, I don't have an example.  If it's a simple approach and we can verify that it doesn't break the main case (one result), then I think that's fine for now.  

It's hard to find persistent test cases when Pubmed is changing the data so frequently.

Reply:
Testing this is a big challenge. I actually ended up doing a major revamp of this in my branch which we can keep on the backburner. Will post ASAP.